doc/service_config.md

```diff
@@ -13,12 +13,20 @@ The service config is a JSON string of the following form:
 ```
 {
   // Load balancing policy name.
-  // Supported values are 'round_robin' and 'grpclb'.
-  // Optional; if unset, the default behavior is pick the first available
-  // backend.
-  // Note that if the resolver returns only balancer addresses and no
-  // backend addresses, gRPC will always use the 'grpclb' policy,
-  // regardless of what this field is set to.
+  // Currently, the only selectable client-side policy provided with gRPC
+  // is 'round_robin', but third parties may add their own policies.
+  // This field is optional; if unset, the default behavior is to pick
+  // the first available backend.
+  // If the policy name is set via the client API, that value overrides
+  // the value specified here.
+  //
+  // Note that if the resolver returns at least one balancer address (as
+  // opposed to backend addresses), gRPC will use grpclb (see
+  // https://github.com/grpc/grpc/blob/master/doc/load-balancing.md),
+  // regardless of what LB policy is requested either here or via the
+  // client API.  However, if the resolver returns backend addresses as
+  // well as balancer addresses, the client may fall back to the requested
+  // policy if it is unable to reach any of the grpclb load balancers.
```

Comment:
> Is the intention that we don't give people the option to stick to pick_first behavior?

No.  People can get pick_first behavior by just not specifying any value here.  We just don't want people to explicitly set this field to a value that is already the default value.

> In contrast, I find the current config language misleading, because the option doesn't have any notion of "fallback" or "secondary" but is in fact **used only when the conditions for the primary choice are not met**. The current spec is disguising the primary/secondary relation as some special-case rules. IMHO, specs should avoid special cases whenever possible, i.e., if they can be instead expressed by more generalized rules. If "fallback" doesn't sound right to you, how about "secondary"?

The core problem here is that there are two cases: (a) grpclb is in use and (b) client-side load balancing is in use.  In case (a), you can indeed look at grpclb as the "primary" mechanism, and the client-side load balancing policy (the value specified here) is a "secondary" or "fallback" mechanism.  But in case (b), the primary mechanism *is* the value specified here, so this is not a fallback.

Adding a term like "fallback" or "secondary" here would slightly clarify things in case (a), but it would be misleading in case (b).  In contrast, not having that term is slightly less clear in case (a) (although I don't think it's actually misleading), but it's fully accurate in case (b).  If we have a choice between being slightly less clear in case (a) or being misleading in case (b), I think the former is a better choice.

From the user's perspective, I think a better model for this behavior is that grpclb is a completely different mechanism than client-side load balancing, and the field in the service config controls only the latter.  We only use client-side load balancing if (1) grpclb is not configured or (2) grpclb is configured but we cannot reach any of the balancers.  I think that's a fairly easy-to-understand description and does not cause any confusion in either case.